File: tests/output/jpa/src/main/javagen/restaurant/jpa_feign/dtos/restaurant/TableItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_feign.dtos.restaurant;

import jakarta.annotation.Generated;

/**
 * Détail d'une table en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface TableItem {

	/**
	 * Identifiant de la table.
	 */
	Integer getId();

	/**
	 * Numéro de la table.
	 */
	String getNumero();

	/**
	 * Capacité de la table (nombre de places).
	 */
	Integer getCapacite();

	/**
	 * Indique si la table est disponible.
	 */
	Boolean getDisponible();

	/**
	 * Restaurant auquel appartient la table.
	 */
	Integer getRestaurantId();

	/**
	 * Identifiant de la table.
	 * @param id value to set.
	 */
	void setId(Integer id);

	/**
	 * Numéro de la table.
	 * @param numero value to set.
	 */
	void setNumero(String numero);

	/**
	 * Capacité de la table (nombre de places).
	 * @param capacite value to set.
	 */
	void setCapacite(Integer capacite);

	/**
	 * Indique si la table est disponible.
	 * @param disponible value to set.
	 */
	void setDisponible(Boolean disponible);

	/**
	 * Restaurant auquel appartient la table.
	 * @param restaurantId value to set.
	 */
	void setRestaurantId(Integer restaurantId);
}

File: tests/output/vanilla-js/model/restaurant/table-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_BOOLEEN, DO_CODE, DO_ID, DO_QUANTITE} from "../../domains";

export interface TableItem {
    id: number;
    numero: string;
    capacite: number;
    disponible: boolean;
    restaurantId: number;
}

export const TableItemEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_ID,
        isRequired: true,
        label: "Id"
    },
    numero: {
        type: "field",
        name: "numero",
        domain: DO_CODE,
        isRequired: true,
        label: "Numero"
    },
    capacite: {
        type: "field",
        name: "capacite",
        domain: DO_QUANTITE,
        isRequired: true,
        label: "Capacite"
    },
    disponible: {
        type: "field",
        name: "disponible",
        domain: DO_BOOLEEN,
        defaultValue: true,
        isRequired: true,
        label: "Disponible"
    },
    restaurantId: {
        type: "field",
        name: "restaurantId",
        domain: DO_ID,
        isRequired: true,
        label: "RestaurantId"
    }
} as const;
